### DISCUSSION TITLE
Pyodide vs. official WebAssembly builds of CPython

### DISCUSSION THREAD

**Discussion by Kodiologist** [pyodide/pyodide#3652] [General]
I recently learned that [the CPython people have been working on builds for WebAssembly](https://pythondev.readthedocs.io/wasm.html), although it sounds to be in a pretty rough state so far, and python.org doesn't seem to offer official builds yet. Are there any plans to integrate some of what Pyodide does upstream into CPython or vice versa?

**Comment by ryanking13** [comment]
Thanks for opening the discussion.

A lot of Pyodide work has been upstream to CPython (#2000), and Pyodide also benefits from CPython WASM support. It is worth noting that Pyodide builds CPython with only minor patches.

Basically, we always try to upstream changes we make in Pyodide to CPython (e.g. [1](https://discuss.python.org/t/a-hook-to-allow-adding-notes-to-modulenotfounderror/21151), [2](https://github.com/pyodide/pyodide/pull/3353)).

**Reply by Kodiologist** [reply]
Makes sense.

**Reply by aliasfoxkde** [reply]
Amazing. Loving Pyodide. Made my own quick and dirty hacky solution with a JS/Node wrapper that simulated pip by installing packages directly from source PyPy packages, inside a web container. It was a pretty cool prototype, but dependencies were a pain, then I found this project. Thanks, and very awesome what you guys are doing!
